File: src/main/java/de/php_perfect/intellij/ddev/actions/DdevStopAction.java
package de.php_perfect.intellij.ddev.actions;

import com.intellij.openapi.project.Project;
import de.php_perfect.intellij.ddev.cmd.Description;
import de.php_perfect.intellij.ddev.state.DdevStateManager;
import de.php_perfect.intellij.ddev.state.State;
import org.jetbrains.annotations.NotNull;

public class DdevStopAction extends DdevRunAction {
    public DdevStopAction() {
        super("stop");
    }

    protected boolean isActive(@NotNull Project project) {
        final State state = DdevStateManager.getInstance(project).getState();

        if (!state.isInstalled()) {
            return false;
        }

        Description description = state.getDescription();

        if (description == null) {
            return false;
        }

        return description.getStatus() == Description.Status.RUNNING || description.getStatus() == Description.Status.UNDEFINED;
    }
}

File: src/main/java/de/php_perfect/intellij/ddev/actions/DdevStopAction.kt
package de.php_perfect.intellij.ddev.actions

import com.intellij.openapi.project.Project
import de.php_perfect.intellij.ddev.cmd.DdevRunner
import de.php_perfect.intellij.ddev.cmd.Description
import de.php_perfect.intellij.ddev.state.DdevStateManager

class DdevStopAction : DdevRunAction() {
    override fun run(project: Project) {
        DdevRunner.getInstance()?.stop(project)
    }

    override fun isActive(project: Project): Boolean {
        val state = DdevStateManager.getInstance(project).getState()

        if (!state.isAvailable() || !state.isConfigured()) {
            return false
        }

        val description = state.getDescription()

        if (description == null) {
            return true
        }

        return description.getStatus() == Description.Status.RUNNING
    }
}
